src/v/storage/tests/batch_consumer_utils_test.cc

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "bytes/iostream.h"
+#include "model/tests/random_batch.h"
+#include "storage/batch_consumer_utils.h"
+#include "storage/directories.h"
+#include "storage/log_manager.h"
+#include "storage/parser.h"
+#include "storage/segment_reader.h"
+#include "storage/tests/utils/disk_log_builder.h"
+#include "test_utils/tmp_dir.h"
+
+#include <seastar/core/io_priority_class.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/defer.hh>
+
+#include <memory>
+
+struct round_robin_consumer : storage::batch_consumer {
+    consume_result
```

Comment:
nit: can you add a comment explaining how this consumes and what is being round-robined?

Reply:
"A consumer that collects every other record batch" or something. Every N?

Reply:
done